native fixes

rurirei opened the pull request:
1. manifest: extractNativeLibs = true
2. native: call SupportSet.Shutdown() when stopService
3. native: call runTun2socks() after vpn.Setup() ~~https://github.com/2dust/v2rayNG/pull/635#pullrequestreview-501540965~~ https://github.com/2dust/v2rayNG/issues/636
4. vpnservice: call sendFd() before tun2socks start
5. native: remove senfFd in Escorting

rurirei committed:
manifest extractNativeLibs = true

rurirei committed:
call SupportSet.Shutdown() when stopService

rurirei committed:
call runTun2socks() after vpn.Setup() succeed

rurirei committed:
remove duplicated Shutdown()

rurirei committed:
call sendFd() before tun2socks start

rurirei committed:
remove senfFd in Escorting

call sendFd() at vpnService.setup(), before tun2socks and wait its start

rurirei committed:
up sendFd try-times to 20

yuhan6665 commented on V2rayNG/app/src/main/AndroidManifest.xml:
```diff
@@ -27,6 +27,7 @@
         android:icon="@mipmap/ic_launcher"
         android:label="@string/app_name"
         android:supportsRtl="true"
+        android:extractNativeLibs="true"
```
Are you sure this is needed?

rurirei replied:
sure. we needs to run binarys inside nativeDir. 
>> When building the release version of your app, package uncompressed .so files in the APK by setting android:extractNativeLibs="false" https://developer.android.com/topic/performance/reduce-apk-size#extract-false

yuhan6665 commented on V2rayNG/app/src/main/kotlin/com/v2ray/ang/service/V2RayVpnService.kt:
```diff
@@ -189,7 +191,7 @@ class V2RayVpnService : VpnService(), ServiceControl {
             } catch (e: Exception) {
                 Log.d(packageName, e.toString())
-                if (tries > 5) break
+                if (tries > 20) break  // below 10 as my experience
```
this comment is confusing

rurirei replied:
sendFd() should only called before tun2socks start, and tries times is below 10 as my experience. call sendFd() after tun2socks started ( inside EscortRun()) causes IOException https://github.com/2dust/v2rayNG/issues/636#issuecomment-703487440 , so 6th commit remove this call.

yuhan6665 commented on AndroidLibV2rayLite/interact.go:
```diff
@@ -171,6 +160,17 @@ func (v *V2RayPoint) pointloop() error {
+		return err
+	}
+
+	log.Printf("EnableLocalDNS: %v\nForwardIpv6: %v\nDomainName: %s",
```
Usually the log statement is at the beginning

rurirei replied:
run tun2socks at the last.

yuhan6665 reviewed (commented):
Generally looks ok

rurirei committed:
update upstream (merge)

* Clean up tun2socks

* Replace divider with standard library

Also, cleanup some unused classes

* Deprecate some anko function

- Replace toast with inline function
- Use standard startActivityForResult

* Replace doasync with Kotlin coroutine

Using a custom coroutine scope, we don't need to keep track of the
running job. We can simply call cancel children for the scope.

* Use standard startActivity

* Replace Anko with support lib AlertDialog

replace some property setter

* Say goodbye to Anko

replace with support lib PreferenceManager
replace LayoutInflater

Co-authored-by: yuhan6665 <1588741+yuhan6665@users.noreply.github.com>
Co-authored-by: 2dust <31833384+2dust@users.noreply.github.com>

rurirei committed:
waits below 2 seconds as my experience

rurirei committed:
force trafficSniffing if localDNS not used.

rurirei committed:
enforce trafficSniffing

yuhan6665 commented on V2rayNG/app/src/main/kotlin/com/v2ray/ang/service/V2RayVpnService.kt:
```diff
@@ -193,7 +194,7 @@ class V2RayVpnService : VpnService(), ServiceControl {
             } catch (e: Exception) {
                 Log.d(packageName, e.toString())
-                if (tries > 20) break  // below 10 as my experience
+                if (tries > 5) break  // waits below 2 seconds as my experience
```
Looks better now

yuhan6665 commented on V2rayNG/app/src/main/kotlin/com/v2ray/ang/util/V2rayConfigUtil.kt:
```diff
@@ -161,11 +161,8 @@ object V2rayConfigUtil {
-                v2rayConfig.inbounds[0].sniffing?.enabled = app.defaultDPreference.getPrefBoolean(SettingsActivity.PREF_SNIFFING_ENABLED, true)
-            }
+            // enforce trafficSniffing to prevent from dns-poisoning
+            v2rayConfig.inbounds[0].sniffing?.enabled = true
```
this doesn't make sense. Sending dns traffic to server also mitigate dns-poisoning. I think someone may not want sniffing function.

rurirei replied:
sniffing had to enabled (for VPN mode), or you will met dns-poisoning, you can try it.

rurirei replied:
https://github.com/2dust/v2rayNG/issues/653

yuhan6665 reviewed (commented):
Please keep the pr commits clean. Use rebase instead of merge. If a previous commit need amend, use force-push to change it. (Instead of adding another revert commit)

rurirei commented:
will clean up later.
